scripts/run_e2e_tests_test.py

```diff
@@ -816,3 +828,30 @@ def test_do_not_run_with_test_suite_not_present_in_suites_list(self):
 
         with self.assertRaisesRegex(SystemExit, '1'):
             run_e2e_tests.main(args=['--suite', 'never'])
+
+    def test_do_not_run_with_test_non_mobile_suite_in_mobile_mode(self):
+        self.exit_stack.enter_context(self.swap_with_checks(
+            run_e2e_tests, 'is_oppia_server_already_running', lambda *_: False))
+        self.exit_stack.enter_context(self.swap_with_checks(
+            run_e2e_tests, 'install_third_party_libraries', lambda _: None,
+            expected_args=[(False,)]))
+        self.exit_stack.enter_context(self.swap_with_checks(
+            run_e2e_tests, 'build_js_files', lambda *_, **__: None,
+            expected_args=[(True,)]))
+        self.exit_stack.enter_context(self.swap_with_checks(
+            servers, 'managed_elasticsearch_dev_server', mock_managed_process))
+        self.exit_stack.enter_context(self.swap_with_checks(
+            servers, 'managed_firebase_auth_emulator', mock_managed_process))
+        self.exit_stack.enter_context(self.swap_with_checks(
+            servers, 'managed_dev_appserver', mock_managed_process))
+        self.exit_stack.enter_context(self.swap_with_checks(
+            servers, 'managed_redis_server', mock_managed_process))
+        self.exit_stack.enter_context(self.swap_with_checks(
+            servers, 'managed_portserver', mock_managed_process))
+        self.exit_stack.enter_context(self.swap_with_checks(
+            servers, 'managed_cloud_datastore_emulator', mock_managed_process))
+        self.exit_stack.enter_context(self.swap(
+            flake_checker, 'check_if_on_ci', lambda: True))
+
+        with self.assertRaisesRegex(SystemExit, '1'):
```

Comment:
Can you make the regex here match more of the error message to make this check more rigorous?

Reply:
I'm not sure how to check for the string output.

Reply:
The second argument to `assertRaisesRegex` is a regular expression that gets compared to the error message. The test passes if the error message contains a match for the regular expression. For example, suppose the error is something like: `SystemExit: The process exited with code 1`. Then your regular expression of `1` would match the `1` in the error message. However, you could have a stronger test by including more of the error message in the regular expression, for example `The process exited with code 1`. Here are the docs for Python regular expressions: https://docs.python.org/3/howto/regex.html

Reply:
@U8NWXD The exit message in the e2e tests are of the form SystemExit: 1, therefore it is not really possible to make this check stronger by adding a string in the assertRaisesRegex check.

Reply:
OK in that case you can assert that the message contains only `1` using the `$` and `^` metacharacters. Here are the docs: https://docs.python.org/3/howto/regex.html#more-metacharacters

Reply:
Done